From 2ae6cf46b591d4aa04ef7d47e7991ed1f461579f Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 28 Aug 2014 03:04:35 -0400
Subject: [acpi] Print diagnostic messages if device links cannot be created

Message-id: <20140828030435.2117.89119.stgit@gir.stowe>
Patchwork-id: 89298
O-Subject: [RHEL7.1 PATCH 10/30] ACPI: Print diagnostic messages if device links cannot be created
Bugzilla: 1128632
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1128632
Upstream Status: 464c114717ae221202ebdbd9aa216035b4626f18

commit 464c114717ae221202ebdbd9aa216035b4626f18
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Thu Aug 8 16:19:19 2013 +0200

    ACPI: Print diagnostic messages if device links cannot be created

    Although the device links created by acpi_bind_one() are not
    essential from the kernel functionality point of view, user space
    may be confused when they are missing, so print diagnostic messages
    to the kernel log if they can't be created.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Acked-by: Toshi Kani <toshi.kani@hp.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/glue.c b/drivers/acpi/glue.c
index f3ead0c..9467229 100644
--- a/drivers/acpi/glue.c
+++ b/drivers/acpi/glue.c
@@ -252,8 +252,15 @@ int acpi_bind_one(struct device *dev, acpi_handle handle)
  acpi_physnode_link_name(physical_node_name, node_id);
  retval = sysfs_create_link(&acpi_dev->dev.kobj, &dev->kobj,
        physical_node_name);
+ if (retval)
+  dev_err(&acpi_dev->dev, "Failed to create link %s (%d)\n",
+   physical_node_name, retval);
+
  retval = sysfs_create_link(&dev->kobj, &acpi_dev->dev.kobj,
        "firmware_node");
+ if (retval)
+  dev_err(dev, "Failed to create link firmware_node (%d)\n",
+   retval);
 
  mutex_unlock(&acpi_dev->physical_node_lock);
 
-- 
1.7.1